src/Storage/ClaimController.ts

```diff
@@ -178,10 +210,37 @@ export class ClaimController {
   }
 
   private downloadEntryClaim = async ({ entry, ...rest }: { entry: Entry }) => {
+    const { ipfsHash } = entry
+    const downloadClaim = async () => {
```

Comment:
Should downloadClaim take ipfsHash as an argument rather than grabbing from lexical scope?

Reply:
I intentionally avoided doing that since this function only exists inside the parent function `downloadEntryClaim`, and `ipfsHash` is an argument of `downloadEntryClaim`. I thought it was just adding noise.

But still I'd like your opinion on this. If you find it more readable and less error prone to explicitly pass the argument, we can try that.

Reply:
Would it allow us to move it outside of the parent and unit test it if we pass it as an argument?

Reply:
@jwicks31 it would require mocking. Which is making me realize I could change the way this is coded so it could be unit tested. I'm thinking something along this lines:

Instead of having

```ts
    const downloadClaim = async (ipfsHash: string) => {
      try {
        return await this.ipfs.cat(ipfsHash)
      } catch (error) {
        if (isFetchTimeoutError(error)) throw new IPFSTimeoutError(ipfsHash)
        else throw new IPFSGenericError(ipfsHash, error)
      }
    }
```

We could have something like 

```ts
const errorToIPFSError = (error, ipfsHash) => isFetchTimeoutError(error)
  ? new IPFSTimeoutError(ipfsHash) 
  : new IPFSGenericError(ipfsHash, error)

const downloadClaim = (ipfsHash: string) =>
  this.ipfs.cat(ipfsHash).catch(errorToIPFSError)
```

In this case we could unit-test `errorToIPFSError`.

Reply:
Alright, got it. I've introduced a new function, Promise.prototype.rethrow, which acts like Promise.catch but throws the result of calling the callback. 

`rethrow` allows us to declare pure error-translating functions like this:

```ts
export const errorToIPFSError = (ipfsHash: string) => (error: Error) =>
  isFetchTimeoutError(error)
    ? new IPFSTimeoutError(ipfsHash)
    : new IPFSGenericError(ipfsHash, error)
```

This function can then be used like this:

```ts
    const downloadClaim = async (ipfsHash: string) =>
      this.ipfs.cat(ipfsHash).rethrow(errorToIPFSError(ipfsHash))
```

I'll update the PR's description with info on Promise.prototype.rethrow.

`errorToIPFSError` is a 100% pure function, no side effects, no throws. It can be easily unit tested.

`downloadClaim` doesn't need to be tested at all, since it's a plain call to two functions which themselves can be tested individually.